nodebuilder/state/keyring.go

```diff
@@ -16,15 +18,20 @@ type AccountName string
 func Keyring(cfg Config, ks keystore.Keystore) (kr.Keyring, AccountName, error) {
 	ring := ks.Keyring()
 	var info *kr.Record
-	// if custom keyringAccName provided, find key for that name
-	if cfg.KeyringAccName != "" {
-		keyInfo, err := ring.Key(cfg.KeyringAccName)
+
+	// go through all keys in the config and check their availability in the KeyStore.
+	for _, accName := range cfg.KeyringKeyNames {
+		keyInfo, err := ring.Key(accName)
 		if err != nil {
-			log.Errorw("failed to find key by given name", "keyring.accname", cfg.KeyringAccName)
+			err = fmt.Errorf("key not found in keystore: %s", accName)
```

Comment:
[blocking]
This might be a different error, not necessarily not found. We should wrap the inner one instead of replacing it. This is a little foot gun that may complicate debugging if something goes wrong with the keyring, especially considering there are multiple backends for them, e.g., Ledger wallet.